azure/datalake/store/utils.py

```diff
@@ -86,7 +86,6 @@ def read_block(f, offset, length, delimiter=None):
         f.seek(start + length)
         seek_delimiter(f, delimiter, 2**16)
         end = f.tell()
-        eof = not f.read(1)
```

Comment:
No side effect that I am aware of. I think this was trying to confirm again that the full chunk was traversed, and it was using the broken logic of read that returned an empty byte array. Since that is all it was doing, there should be no side effect. 

I will be adding some more in depth tests for delimiter validation in the future.